src/components/SidePanels/OutlinerPanel/OutlinerTree.tsx

```diff
@@ -314,6 +314,14 @@ const RegionControls: FC<RegionControlsProps> = observer(({
     }
   }, [item, item?.toggleHidden, hidden]);
 
+  const onToggleCollapsed = useCallback((e: MouseEvent) => {
+    onToggleCollapsed(e);
+  }, [toggleCollapsed]);
+
+  const onToggleLocked = useCallback(() => {
+    item.setLocked(!item.locked);
+  }, [item]);
```

Comment:
maybe `item.locked` as dependency? or even `item.setLocked(someFn)` to always use current value.

Reply:
Fixed